kani-compiler/src/kani_middle/reachability.rs

```diff
@@ -518,22 +422,22 @@ impl<'a, 'tcx> MirVisitor<'tcx> for MonoItemsFnCollector<'a, 'tcx> {
                                 // necessarily true. It could be any argument or
                                 // even the return type, for instance for a
                                 // trait like `FromIterator`.
-                                let generic_ty = outer_args[0].ty(self.body, tcx).peel_refs();
-                                let receiver_ty = tcx.instantiate_and_normalize_erasing_regions(
-                                    substs,
-                                    ParamEnv::reveal_all(),
-                                    EarlyBinder::bind(generic_ty),
-                                );
+                                let receiver_ty = args.0[0].expect_ty();
+                                trace!(?receiver_ty, ?callee, "**** receiver:");
```

Comment:
Looks much simpler than what we used to do. Do we want to keep the `trace!` call?

Reply:
No. Also a debug artifact that I need to remove. :)